pkg/storage/business_case.go

```diff
@@ -35,7 +35,7 @@ func (s *Store) FetchBusinessCaseByID(id uuid.UUID) (*models.BusinessCase, error
 			business_case
 			LEFT JOIN estimated_lifecycle_cost ON business_case.id = estimated_lifecycle_cost.business_case
 		WHERE
-			business_case.id = $1
+			business_case.id = $1 AND business_case.status != 'ARCHIVED'
```

Comment:
I have some concerns about the longevity of the API effectively saying "I know nothing about this so called business case that I secretly do have in my DB.", but also I recognize that can be a later-us problem.